lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -7898,37 +7903,131 @@ namespace Js
     }
 
 #if ENABLE_NATIVE_CODEGEN
-    void EntryPointInfo::AddWeakFuncRef(RecyclerWeakReference<FunctionBody> *weakFuncRef, Recycler *recycler)
+    DWORD_PTR EntryPointInfo::GetNativeAddress() const
+    {
+        // need the assert to skip for asmjsFunction as nativeAddress can be interpreter too for asmjs
+        Assert(this->GetState() == CodeGenRecorded || this->GetState() == CodeGenDone || this->isAsmJsFunction);
+
+        // !! this is illegal, however (by design) `IsInNativeAddressRange` (right above) needs it
+        return reinterpret_cast<DWORD_PTR>(this->GetNativeEntryPointData()->GetNativeAddress());
+    }
+
+    Js::JavascriptMethod EntryPointInfo::GetThunkAddress() const
+    {
+        Assert(this->GetState() == CodeGenRecorded || this->GetState() == CodeGenDone);
+
+        return this->GetNativeEntryPointData()->GetThunkAddress();
+    }
+
+    Js::JavascriptMethod EntryPointInfo::GetNativeEntrypoint() const
     {
-        Assert(this->state == CodeGenPending);
+        Assert(this->GetState() == CodeGenRecorded || this->GetState() == CodeGenDone || this->isAsmJsFunction);
```

Comment:
Not your code but what is the model for asmJs functions? How can there be a native entry point if we havent done codegen? @Cellule might know?